extensions/Worker.Extensions.Http.AspNetCore.Analyzers/src/CodeFixForHttpResultAttributeExpected.cs

```diff
@@ -0,0 +1,93 @@
+        {
+            Diagnostic diagnostic = context.Diagnostics.First();
+            context.RegisterCodeFix(new AddHttpResultAttribute(context.Document, diagnostic), diagnostic);
+
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// CodeAction implementation which adds the HttpResultAttribute on the return type of a function using the multi-output bindings pattern.
+        /// </summary>
+        private sealed class AddHttpResultAttribute : CodeAction
+        {
+            private readonly Document _document;
+            private readonly Diagnostic _diagnostic;
+            private readonly string ExpectedAttributeName = "HttpResultAttribute";
+
+
+            internal AddHttpResultAttribute(Document document, Diagnostic diagnostic)
+            {
+                this._document = document;
+                this._diagnostic = diagnostic;
+            }
+
+            public override string Title => "Add HttpResultAttribute";
+
+            public override string EquivalenceKey => null;
+
+            /// <summary>
+            /// Returns an updated Document with HttpResultAttribute added to the relevant property.
+            /// </summary>
+            protected override async Task<Document> GetChangedDocumentAsync(CancellationToken cancellationToken)
+            {
+                // Get the syntax root of the document
+                var root = await _document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var semanticModel = await _document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                var typeNode = root.FindNode(this._diagnostic.Location.SourceSpan)
+                    .FirstAncestorOrSelf<TypeSyntax>();
+
+                var typeSymbol = semanticModel.GetSymbolInfo(typeNode).Symbol;
+                var typeDeclarationSyntaxReference = typeSymbol.DeclaringSyntaxReferences.FirstOrDefault();
+                if (typeDeclarationSyntaxReference == null)
+                {
+                    return _document;
+                }
+
+                var typeDeclarationNode = await typeDeclarationSyntaxReference.GetSyntaxAsync(cancellationToken);
+
+                var propertyNode = typeDeclarationNode.DescendantNodes()
+                    .OfType<PropertyDeclarationSyntax>()
+                    .First(prop =>
+                    {
+                        var propertyType = semanticModel.GetTypeInfo(prop.Type).Type;
+                        return propertyType != null && propertyType.Name == "IActionResult";
+                    });
+
+                var attribute = SyntaxFactory.Attribute(SyntaxFactory.IdentifierName(ExpectedAttributeName));
+
+                var newPropertyNode = propertyNode
+                    .AddAttributeLists(SyntaxFactory.AttributeList(SyntaxFactory.SingletonSeparatedList(attribute)));
+
+                var newRoot = root.ReplaceNode(propertyNode, newPropertyNode);
```

Comment:
Still working on getting this replacement to have the correct indentation - this is why one test is still failing.